src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -94,7 +94,7 @@ protected function instantiateController($class)
         } catch (\TypeError $e) {
         }
 
-        if ($this->container instanceof Container && in_array($class, $this->container->getRemovedIds(), true)) {
+        if ($this->container instanceof Container && isset($this->container->getRemovedIds()[strtolower($class)])) {
```

Comment:
that's a bit unrelated, but still found while playing with getRemovedIds: removed services are provided as *keys*

Reply:
are you sure about the strtolower ? The normalized id is not the lowercase one

Reply:
indeed, fixed